pkg/ebpf/c/tracee.bpf.c

```diff
@@ -6380,3 +6406,812 @@ int tc_ingress(struct __sk_buff *skb)
+
+// inodemap
+
+typedef struct net_task_context {
+    struct task_struct *task;
+    task_context_t taskctx;
+} net_task_context_t;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_LRU_HASH);
+    __uint(max_entries, 10240);             // simultaneous sockets being traced
+    __type(key, u64);                       // socket inode number ...
+    __type(value, struct net_task_context); // ... linked to a task context
+} inodemap SEC(".maps");                    // relate sockets and tasks
+
+// entrymap
+
+typedef struct entry {
+    long unsigned int args[6];
+} entry_t;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_LRU_HASH);
+    __uint(max_entries, 1024);       // simultaneous tasks being traced for entry/exit
+    __type(key, u32);                // host thread group id (tgid or tid) ...
+    __type(value, struct entry);     // ... linked to entry ctx->args
+} entrymap SEC(".maps");             // can't use args_map (indexed by existing events only)
+
+// clang-format on
+
+//
+// support functions for network code
+//
+
+static __always_inline u64 sizeof_net_event_context_t(void)
+{
+    return sizeof(net_event_context_t) - sizeof(net_event_contextmd_t);
+}
+
+static __always_inline void set_net_task_context(event_data_t *data, net_task_context_t *netctx)
+{
+    netctx->task = data->task;
+    __builtin_memcpy(&netctx->taskctx, &data->context.task, sizeof(task_context_t));
+}
+
+static __always_inline void should_submit_net_event(net_event_context_t *neteventctx,
+                                                    config_entry_t *config)
+{
+    // configure network events that should be sent to userland
+
+    if (should_submit(NET_PACKET_IP, config))
+        neteventctx->md.submit |= SUB_NET_PACKET_IP;
+
+    if (should_submit(NET_PACKET_TCP, config))
+        neteventctx->md.submit |= SUB_NET_PACKET_TCP;
+
+    if (should_submit(NET_PACKET_UDP, config))
+        neteventctx->md.submit |= SUB_NET_PACKET_UDP;
+
+    if (should_submit(NET_PACKET_ICMP, config))
+        neteventctx->md.submit |= SUB_NET_PACKET_ICMP;
+
+    if (should_submit(NET_PACKET_ICMPV6, config))
+        neteventctx->md.submit |= SUB_NET_PACKET_ICMPV6;
+
+    if (should_submit(NET_PACKET_DNS, config))
+        neteventctx->md.submit |= SUB_NET_PACKET_DNS;
+}
+
+//
+// Socket Creation: keep track of created socket inodes per traced task
+//
+
+SEC("kprobe/sock_alloc_file")
+int BPF_KPROBE(trace_sock_alloc_file)
+{
+    // runs every time a socket is created (entry)
+
+    event_data_t data = {};
+
+    if (!init_event_data(&data, ctx))
+        return 0;
+
+    if (!should_trace(&data))
+        return 0;
+
+    struct entry entry = {0};
+
+    // save args for retprobe
+    entry.args[0] = PT_REGS_PARM1(ctx); // struct socket *sock
+    entry.args[1] = PT_REGS_PARM2(ctx); // int flags
+    entry.args[2] = PT_REGS_PARM2(ctx); // char *dname
+
+    struct socket *sock = (void *) PT_REGS_PARM1(ctx);
+
+    if (!is_socket_supported(sock))
+        return 0;
```

Comment:
An optimization returning early, perhaps?

```suggestion
    struct socket *sock = (void *) PT_REGS_PARM1(ctx);
    if (!is_socket_supported(sock))
        return 0;

    // save args for retprobe
    entry.args[0] = (unsgined long) sock; // struct socket *sock
    entry.args[1] = PT_REGS_PARM2(ctx);   // int flags
    entry.args[2] = PT_REGS_PARM2(ctx);   // char *dname
```